### PR TITLE
Refactoring of the Dapp Registry

### PR DESCRIPTION
**Pull request by ngotchac**
Refactoring of the Dapp Registry built-in dapp.

No need to register the Manifest/Image/Content URL in GithubHint dapp anymore, the Dapp Registry is taking care of this now (thus, is becoming a secure dapp).

New UI as well.

Stil todo:

- [x] Better error display on editing + show transaction**s** progress
- [x] Delete a dapp
- [x] Add a new dapp
- [x] Fill dapp meta-data from the Registry (#4342)

image

image

### CODE REVIEW

**Comment by jacogr** [comment]
Wishlist -

1. Dapp registration via URL (opens in extension/built-in browser)
2. Auto-populate of meta information from manifest.json (e.g. main URL entered, manifest.json & image auto detected & populated)
3. ...

File: js/src/dapps/dappreg/Application/application.css [hunk 1]
```diff
@@ -15,10 +15,12 @@
 /* along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+@import '../_colors.css';
+
 .body {
   color: #333;
```
Comment (jacogr) [inline js/src/dapps/dappreg/Application/application.css]: Probably would have made sense to put this colour in `colors.css` as well. (Minor)

File: js/src/dapps/dappreg/CreateDappCard/createDappCard.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import PlusImage from '~/../assets/images/dapps/plus.svg';
+
+export default class CreateDappCard extends Component {
+  state = {
+    dappId: null,
+    focus: false,
+    open: false
+  };
+
+  dappsStore = DappsStore.instance();
```
Comment (jacogr) [inline js/src/dapps/dappreg/CreateDappCard/createDappCard.js]: We generally use `.get()`

File: js/src/dapps/dappreg/Modal/modal.css [hunk 3]
```diff
@@ -15,95 +15,120 @@
...
-  .body {
-    bottom: 0;
-    left: 0;
-    position: fixed;
-    right: 0;
-    top: 0;
-    text-align: center;
-    z-index: 50;
+  align-items: center;
+  background-color: rgba(40, 40, 40, 0.75);
```
Comment (jacogr) [inline js/src/dapps/dappreg/Modal/modal.css]: Also would have put this in `colors.css` (There are a couple, not commenting on all). Not a crisis where there are no re-use, bust consistency for all.

File: js/src/dapps/dappreg/Transactions/transactions.css [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  .header {
+    flex: 0 0 auto;
+    height: 1.5rem;
+    position: relative;
+    width: 100%;
+
+    > * {
+      display: inline-block;
+      font-family: 'Roboto Mono', monospace;
+      font-size: 0.8rem;
```
Comment (jacogr) [inline js/src/dapps/dappreg/Transactions/transactions.css]: Slightly different from the 0.75rem used elsewhere. (Although this is inside a header.)

File: js/src/dapps/dappreg/_colors.css [hunk 5]
```diff
@@ -15,5 +15,4 @@
 /* along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-.app {
-}
+$blue: rgb(41, 128, 185);
```
Comment (jacogr) [inline js/src/dapps/dappreg/_colors.css]: Would have made the name "title-background" or "title-color" (something specific, not tied to the color but actual usage thereof)

File: js/src/dapps/dappreg/DappModal/dappModal.js [hunk 6]
```diff
@@ -0,0 +1,412 @@
...
+        <div className={ styles.actions }>
+          <Button
+            className={ styles.button }
+            disabled={ !dapp.canSave }
+            label='Save'
+            onClick={ this.handleSave }
+          />
+          <Button
+            className={ styles.button }
+            label='Fetch Registry'
```
Comment (jacogr) [inline js/src/dapps/dappreg/DappModal/dappModal.js]: I find this a bit confusing when I looked at the UI. "Reset" maybe?

**Comment by jacogr** [comment]
Couple of minor comments. My only grumble is that we lost the ability for the contract owner to remove apps from the dapp. (e.g. the contract owner account should have the ability to delete any dapp - existing in current version, not in this one - it can never edit them, but can remove any app)

Huge step forward, I really like it as a base to build on now.

File: js/src/dapps/dappreg/DappModal/dappModal.js (hunk 6 above)
Comment (ngotchac) [inline js/src/dapps/dappreg/DappModal/dappModal.js]: So this is in fact for  #4342
It fetches the registered IMG and CONTENT hashes from the registry, based on the selected dapp owner : `owner address ==(registry reverse)==> registry name ==(registry lookup)==> [ IMG_GHH_hash ; CONTENT_GHH_hash ] ==> fill the form`.

So it's not a reset, it's fetching data from the registry, based on the selected owner address. Couldn't think of a better explicit name though :/

**Comment by ngotchac** [comment]
Ok, so I addressed the PR grumbles, and added the ability for the dapp reg contract owner to delete dapps.

File: js/src/dapps/dappreg/DappModal/dappModal.js (hunk 6 above)
Comment (jacogr) [inline js/src/dapps/dappreg/DappModal/dappModal.js]: Ahhh, that one. Ok, will ponder a bit on a better more explicit name, it escapes me at present as well...

**Pull request merged**